test/unit/client/creating_resource_with_meta_test.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:json_api/client.dart';
+import 'package:json_api/document.dart';
+import 'package:json_api/http.dart';
+import 'package:test/test.dart';
+
+import '../../helper/test_http_handler.dart';
+
+void main() async {
+  test('request actually has the meta property', () async {
+    final handler = TestHttpHandler();
+    final client = JsonApiClient(handler);
+
+    final uri = Uri.parse('https://github.com/f3ath/json-api-dart');
+    final person =
+        Resource('people', '123', attributes: {'name': 'Te Cheng Hung'});
+    final meta = {'friend': 'Martin Fowler'};
+
+    handler.nextResponse = HttpResponse(500, body: 'Something went wrong');
```

Comment:
We're testing a successful case, so why use an error response? Why don't we use a simple 201 with no body?

Reply:
You're right, I will fix that, thanks!